plugins/modules/ec2_scaling_policy.py

```diff
@@ -6,178 +6,380 @@
-            min_adjustment_step=min_adjustment_step,
-            cooldown=cooldown)
+    changed = False
+    asg_name = module.params['asg_name']
+    policy_type = module.params['policy_type']
+    policy_name = module.params['name']
 
-        try:
-            connection.create_scaling_policy(sp)
-            policy = connection.get_all_policies(as_group=asg_name, policy_names=[sp_name])[0]
-            module.exit_json(changed=True, name=policy.name, arn=policy.policy_arn, as_name=policy.as_name, scaling_adjustment=policy.scaling_adjustment,
-                             cooldown=policy.cooldown, adjustment_type=policy.adjustment_type, min_adjustment_step=policy.min_adjustment_step)
-        except BotoServerError as e:
-            module.fail_json(msg=str(e))
-    else:
-        policy = scalingPolicies[0]
-        changed = False
+    params = dict(PolicyName=policy_name,
+                  PolicyType=policy_type,
+                  AutoScalingGroupName=asg_name,
+                  AdjustmentType=module.params['adjustment_type'])
 
-        # min_adjustment_step attribute is only relevant if the adjustment_type
-        # is set to percentage change in capacity, so it is a special case
-        if getattr(policy, 'adjustment_type') == 'PercentChangeInCapacity':
-            if getattr(policy, 'min_adjustment_step') != module.params.get('min_adjustment_step'):
-                changed = True
+    # min_adjustment_step attribute is only relevant if the adjustment_type
+    # is set to percentage change in capacity, so it is a special case
+    if module.params['adjustment_type'] == 'PercentChangeInCapacity':
+        if module.params['min_adjustment_step']:
+            params['MinAdjustmentMagnitude'] = module.params['min_adjustment_step']
+
+    if policy_type == 'SimpleScaling':
+        # can't use required_if because it doesn't allow multiple criteria -
+        # it's only required if policy is SimpleScaling and state is present
+        if not module.params['scaling_adjustment']:
+            module.fail_json(msg='scaling_adjustment is required when policy_type is SimpleScaling '
+                             'and state is present')
+        params['ScalingAdjustment'] = module.params['scaling_adjustment']
+        if module.params['cooldown']:
+            params['Cooldown'] = module.params['cooldown']
+
+    if policy_type == 'StepScaling':
+        if not module.params['step_adjustments']:
+            module.fail_json(msg='step_adjustments is required when policy_type is StepScaling '
+                             'and state is present')
+        params['StepAdjustments'] = []
+        for step_adjustment in module.params['step_adjustments']:
+            step_adjust_params = dict(ScalingAdjustment=step_adjustment['scaling_adjustment'])
+            # Although empty bounds are allowed, not setting the 0 end of the bound seems
+            # to cause problems in the UI. We'll leave the +/- infinity end unset if not set
```

Comment:
As best I can tell 0 and None both have the effect of setting infinity in the console, as does entirely commenting out lines 281, 282, 285, and 286 (the "= 0 if unset" bits).  Where did you run into issues?

I'm using https://github.com/ansible-collections/community.aws/blob/26eaf02cbec4ed2ed3a4681fea375af7522d83d7/tests/integration/targets/ec2_scaling_policy/tasks/main.yml#L145-L161 as a test followed by a pause so I can poke around the console.

Reply:
This was Will's work, it's possible that Amazon have updated their UIs to fix the issue he was referring to.

Reply:
I've removed the work-arounds and we can handle the issue as a bug (or raise a case with Amazon) if someone encounters the issue.